File: Arrays_Hashing/ValidSudoku/ValidSudoku.swift
class Solution {
    func isValidSudoku(_ board: [[Character]]) -> Bool {
        var rows = [[Int: Bool]](repeating: [:], count: 9)
        var cols = [[Int: Bool]](repeating: [:], count: 9)
        var squares = [[[Int: Bool]]](repeating: [[Int: Bool]](repeating: [:], count: 3), count: 3)

        for r in 0..<9 {
            for c in 0..<9 {
                let cell = board[r][c]
                guard cell != "." else { continue }

                let num = Int(String(cell))!

                if rows[r][num] == true { return false }
                if cols[c][num] == true { return false }
                if squares[r/3][c/3][num] == true { return false }

                rows[r][num] = true
                cols[c][num] = true
                squares[r/3][c/3][num] = true
            } 
        }
        return true
    }
}

class Solution {
    func isValidSudoku(_ board: [[Character]]) -> Bool {
        var rows = [Set<Int>](repeating: [], count: 9)
        var cols = [Set<Int>](repeating: [], count: 9)
        var squares = [[Set<Int>]](repeating: [Set<Int>](repeating: [], count: 3), count: 3)

        for r in 0..<9 {
            for c in 0..<9 {
                let cell = board[r][c]
                guard cell != "." else { continue }

                let num = Int(String(cell))!

                if rows[r].contains(num) { return false }
                if cols[c].contains(num) { return false }
                if squares[r/3][c/3].contains(num) { return false }

                rows[r].insert(num)
                cols[c].insert(num)
                squares[r/3][c/3].insert(num)
            }
        }

        return true
    }
}

File: Arrays_Hashing/ValidSudoku/ValidSudoku.kt

class Solution {
    fun isValidSudoku(board: Array<CharArray>): Boolean {
       	val rows = Array(9) { HashSet<Int>() }
val cols = Array(9) { HashSet<Int>() }
val squares = Array(3) { Array(3) { HashSet<Int>() } }

                	
         for (r in 0..<9) {
             for (c in 0..<9) {
    			 val cell = board[r][c]
                 if (cell == '.') continue
                 val num = cell.digitToInt()
                 
                 if (rows[r].contains(num)) return false
                 if (cols[c].contains(num)) return false
                 if (squares[r / 3][c / 3].contains(num)) return false

                 rows[r].add(num)
                 cols[c].add(num)
                 squares[r / 3][c / 3].add(num)
                 
             }
         }
        
        
     	
        return true
        
        
    }
}
